server/game/cards/characters/01/oldforesthunter.js

```diff
@@ -4,30 +4,16 @@ class OldForestHunter extends DrawCard {
     setupCardAbilities(ability) {
         this.action({
             title: 'Discard a card to gain 1 gold',
-            method: 'discard',
-            limit: ability.limit.perPhase(1)
+            condition: () => !this.controller.cannotGainGold,
+            cost: ability.costs.discardFromHand(),
+            limit: ability.limit.perPhase(1),
+            handler: context => {
+                this.game.addGold(this.controller, 1);
```

Comment:
Agreed, it is probably preferable to see for example 'DukeTax uses Tyrion Lannister to gain 2 gold' followed by 'DukeTax cannot gain gold', instead of the other way around. 

In these three instances it doesn't matter though since these cannot prompt anyway if you cannot gain gold (and they make up probably >95% of uses cases for gaining gold during the challenge phase).